### ISSUE TITLE
warn The gatsby-plugin-mdx plugin has generated no Gatsby nodes. Do you need it?

### ISSUE THREAD

**Issue by AlexEyler** [mrmartineau/gatsby-theme-code-notes#35]
The above error message results in a build failure when we try to get query for the 'fields' column in graphQL:

```
6:10:31 PM: error There was an error in your GraphQL query:
6:10:31 PM: Cannot query field "fields" on type "Mdx".
6:10:31 PM: If you don't expect "fields" to exist on the type "Mdx" it is most likely a typo.
However, if you expect "fields" to exist there are a couple of solutions to common problems:
6:10:31 PM: - If you added a new data source and/or changed something inside gatsby-node.js/gatsby-config.js, please try a restart of your development server
6:10:31 PM: - The field might be accessible in another subfield, please try your query in GraphiQL and use the GraphiQL explorer to see which fields you can query and what shape they have
6:10:31 PM: - You want to optionally use your field "fields" and right now it is not used anywhere. Therefore Gatsby can't infer the type and add it to the GraphQL schema. A quick fix is to add a least one entry with that field ("dummy content")
6:10:31 PM: It is recommended to explicitly type your GraphQL schema if you want to use optional fields. This way you don't have to add the mentioned "dummy content". Visit our docs to learn how you can define the schema for "Mdx":
```

I believe the fix is to activate `gatsby-source-filesystem` before activating `gatsby-plugin-mdx` (see https://github.com/gatsbyjs/gatsby/issues/15594). Here's a change in my repo I made to fix the issue: https://github.com/AlexEyler/my-notes/commit/936385f7a5841e264b9efce83cab3eccfab0c3da

I'll create a PR for this fix.

**Comment by mrmartineau** [comment]
Thanks for raising this and indeed fixing it 👍

**Comment by mrmartineau** [comment]
@AlexEyler your fix was just released: [v1.14.1](https://github.com/mrmartineau/gatsby-theme-code-notes/releases/tag/v1.14.1)

**Comment by AlexEyler** [comment]
No worries, thanks!
